snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/tracker/Tracker.java

```diff
@@ -632,6 +634,7 @@ public void track(final @NonNull Event event) {
             this.emitter.add(payload);
             event.endProcessing(this);
         });
+        return trackerEvent.eventId;
```

Comment:
👍 I think in the future we could remove this asynchronous part about event processing. It shouldn't be so expensive in my opinion. Even because on iOS it's not async. I've always thought to make the iOS processing async like on Android, but now I'm more keen to go in the opposite direction, making Android event processing synchrnous until the payload reaches the EventStore.
Just a note for future brainstorming.